src/BeamAdapter/component/BeamInterpolation.inl

```diff
@@ -555,10 +559,19 @@ void BeamInterpolation<DataTypes>::getNumberOfCollisionSegment(Real &dx, unsigne
 }
 
 template <class DataTypes>
-void BeamInterpolation<DataTypes>::getYoungModulusAtX(int beamId,Real& /*x_curv*/, Real& youngModulus, Real& cPoisson)
+void BeamInterpolation<DataTypes>::getYoungModulusAtX(int beamId, Real& /*x_curv*/, Real& youngModulus, Real& cPoisson)
 {
-    youngModulus = Real(d_defaultYoungModulus.getValue()[beamId]);
-    cPoisson     = Real(d_poissonRatio.getValue()[beamId]);
+    if (beamId < d_defaultYoungModulus.getValue().size()) {
+        youngModulus = d_defaultYoungModulus.getValue()[beamId];
+    } else {
+        youngModulus = m_defaultYoungModulus;
+    }
+
+    if (beamId < d_poissonRatio.getValue().size()) {
+        cPoisson     = d_poissonRatio.getValue()[beamId];
+    } else {
+        cPoisson     = m_defaultPoissonRatio;
+    }
```

Comment:
```suggestion
    const auto& defaultYoungModuli = d_defaultYoungModulus.getValue();
    if (beamId < defaultYoungModuli .size()) {
        youngModulus = defaultYoungModuli[beamId];
    } else {
        youngModulus = m_defaultYoungModulus;
    }
    
    const auto& poissonRatios = d_poissonRatio.getValue();
    if (beamId < poissonRatios.size()) {
        cPoisson     = poissonRatios[beamId];
    } else {
        cPoisson     = m_defaultPoissonRatio;
    }
```

Maybe the optimizer is doing that by itself but in case of...